addons/payment/models/payment_acquirer.py

```diff
@@ -1063,29 +1063,34 @@ def form_feedback(self, data, acquirer_name):
     # --------------------------------------------------
 
     def s2s_do_transaction(self, **kwargs):
+        self.flush()
         custom_method_name = '%s_s2s_do_transaction' % self.acquirer_id.provider
         for trans in self:
             trans._log_payment_transaction_sent()
             if hasattr(trans, custom_method_name):
                 return getattr(trans, custom_method_name)(**kwargs)
```

Comment:
@AntoineVDV what about adding an explicit `cr.commit()` in the actual S2S method, or just before invoking it?